### ISSUE TITLE
Can't refill vehicle with water or clean water

### ISSUE THREAD

**Issue by MonkWho** [CleverRaven/Cataclysm-DDA#24887]
When trying to refill vehicle tanks with water or clean water it says "You don't have the necessary item at hand" even if you have containers with said liquid in your inventory or in containers nearby. Refilling with gasoline or diesel works as expected.
My vehicle also has a kitchen unit installed. I can drink clean water from it. But if I boil water using the kitchen I can't put new clean water into a vehicle tank (empty or with existing liquid in it).
OS: Windows 10
Game Version: 0.C-30616-g5d18af2

To Reproduce:
 - Examine the vehicle
 - Try to re[f]ill a tank that has clean water in it

Ran the same test in 0.C-30507-g85b8217 (my "previous_version" in the launcher) and it works like a charm with water and clean water.

**Comment by MonkWho** [comment]
Did some quick checking and it looks like this problem starts in #7701. If that helps to track the cause.

**Comment by Androwski3000** [comment]
Can this be solved by installing a water faucet? - I´m just in the INSTALL menu of a RV and you can install external tanks, like a 200 L tank with a steel barrel. In the description text it says, install a water faucet, if you keep water in that tank.

I´ll try it and report. The next water faucet I know of is miles away right now.

**Comment by MonkWho** [comment]
Tested with water faucet as well. Does not work. It feel like some change was made that sees gasoline and diesel as an acceptable fluid to fill vehicle tanks but does not see water or clean water.

**Comment by paulenka-aleh** [comment]
I guess freezing mechanics. Now ambient temperature makes water active. And active liquids cannot be in vehicle...

**Comment by MonkWho** [comment]
So do we need to put an extra flag on non perishable liquids (water, alcohol, bleach, etc) to distinguish them from "those that rot" active ones?

**Comment by narc0tiq** [comment]
> I was able to refill the vehicle tanks before build 7701.

Yes, because #24228 (among other things) expanded the active items to be all comestibles, regardless of rot-ability, heat, or anything else. It's the same issue, just much harder to mitigate now.

**Comment by MonkWho** [comment]
@narc0tiq, that makes sense. I guess for now we have to carry non fuel liquids in tanks in storage and not attached to the vehicle eh? Thank you for explaining it. Hope someone figures out how to handle this soon.
